syntax = "proto3";

package analyzer;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service AnalyzerService {
  rpc GetTargetDetails (GetTargetDetailsRequest) returns (GetTargetDetailsResponse);
  rpc ListTargets (ListTargetsRequest) returns (ListTargetsResponse);
  rpc DeleteTarget (GetTargetDetailsRequest) returns (google.protobuf.Empty);
}

message TargetSummary {
  string                    id         = 1;
  string                    url        = 2;
  string                    status     = 3;
  google.protobuf.Timestamp checked_at = 4;
  string                    comment    = 5;
  google.protobuf.Timestamp created_at = 6;
}

message TargetDetails {
  message HistoryItem {
    string                    status     = 1;
    google.protobuf.Timestamp checked_at = 2;
    string                    comment    = 3;
  }

  string                    id         = 1;
  string                    url        = 2;
  google.protobuf.Timestamp created_at = 3;
  repeated HistoryItem      history    = 4;

}

message GetTargetDetailsRequest {
  string id = 1;
}

message GetTargetDetailsResponse {
  TargetDetails target = 1;
}

message ListTargetsRequest {
  int32 limit  = 1;
  int32 offset = 2;
}

message ListTargetsResponse {
  repeated TargetSummary targets = 1;
}

message DeleteTargetRequest {
  string id = 1;
}
